File: LEETCODE/1. 两数之和.java
class MySolution {
    public int[] twoSum(int[] nums, int target) {
        int length = nums.length;
        int result[] = new int[2];
        loop:
        for(int i=0;i<length;i++) {
            for(int j=0;j<length;j++) {
                if(nums[i]+nums[j]==target&&j!=i) {
                    result[0] = i;
                    result[1] = j;
                    break loop;
                }
            }
            
        }
        return result;
    }
}

class OthersSolution{
    public int[] twoSum(int[] nums,int target) {
        int result[] = new int[2];
        if(nums==null||nums.length<2)
            return result;
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
        for(int i=0;i<nums.length;i++) {
            if(!map.containsKey(target - nums[i])) {
                map.put(nums[i], i);
            }else {
                result[0] = map.get(target - nums[i]);
                result[1] = i;
                break;
            }
        }
        return result;
    }
}

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int dvalue = target - nums[i];
            if (map.containsKey(dvalue)){
                return new int[]{map.get(dvalue), i};
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumentException("No pairs of number");
    }
}

File: LEETCODE/kotlin/1. 两数之和.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int,Int>()
        for(i in nums.indices) {
            val diff = target - nums[i]
            if(map.containsKey(diff)) {
                //!!表示相信这个值非空，不加!!会报错
                return intArrayOf(map[diff]!!, i)
            }
            //kotlin不推荐使用put和get
            map[nums[i]] = i
        }
        return intArrayOf(-1, -1)
    }
}